src/client/v2/algod/models/types.ts

```diff
@@ -0,0 +1,1390 @@
+    this.buildNumber = buildNumber;
+    this.channel = channel;
+    this.commitHash = commitHash;
+    this.major = major;
+    this.minor = minor;
+
+    this.attribute_map = {
+      branch: 'branch',
+      buildNumber: 'build_number',
+      channel: 'channel',
+      commitHash: 'commit_hash',
+      major: 'major',
+      minor: 'minor',
+    };
+  }
+}
+
+/**
+ *
+ */
+export class CatchpointAbortResponse extends BaseModel {
+  /**
+   * Creates a new `CatchpointAbortResponse` object.
+   * @param catchupMessage - Catchup abort response string
+   */
+  constructor(public catchupMessage: string) {
+    super();
+    this.catchupMessage = catchupMessage;
+
+    this.attribute_map = {
+      catchupMessage: 'catchup-message',
+    };
+  }
+}
+
+/**
+ *
+ */
+export class CatchpointStartResponse extends BaseModel {
+  /**
+   * Creates a new `CatchpointStartResponse` object.
+   * @param catchupMessage - Catchup start response string
+   */
+  constructor(public catchupMessage: string) {
+    super();
+    this.catchupMessage = catchupMessage;
+
+    this.attribute_map = {
+      catchupMessage: 'catchup-message',
+    };
+  }
+}
+
+/**
+ * Teal compile Result
+ */
+export class CompileResponse extends BaseModel {
+  /**
+   * Creates a new `CompileResponse` object.
+   * @param hash - base32 SHA512_256 of program bytes (Address style)
+   * @param result - base64 encoded program bytes
+   */
+  constructor(public hash: string, public result: string) {
+    super();
+    this.hash = hash;
+    this.result = result;
+
+    this.attribute_map = {
+      hash: 'hash',
+      result: 'result',
+    };
+  }
+}
+
+/**
+ * Request data type for dryrun endpoint. Given the Transactions and simulated
+ * ledger state upload, run TEAL scripts and return debugging information.
+ */
+export class DryrunRequest extends BaseModel {
+  public accounts: Account[];
+  public apps: Application[];
+  public latestTimestamp: number;
+  public protocolVersion: string;
+  public round: bigint;
+  public sources: DryrunSource[];
+  public txns: EncodedSignedTransaction[];
+
+  /**
+   * Creates a new `DryrunRequest` object.
+   * @param accounts -
+   * @param apps -
+   * @param latestTimestamp - LatestTimestamp is available to some TEAL scripts. Defaults to the latest
+   * confirmed timestamp this algod is attached to.
+   * @param protocolVersion - ProtocolVersion specifies a specific version string to operate under, otherwise
+   * whatever the current protocol of the network this algod is running in.
+   * @param round - Round is available to some TEAL scripts. Defaults to the current round on the
+   * network this algod is attached to.
+   * @param sources -
+   * @param txns -
+   */
+  constructor({
+    accounts,
+    apps,
+    latestTimestamp,
+    protocolVersion,
+    round,
+    sources,
+    txns,
+  }: {
+    accounts: Account[];
+    apps: Application[];
+    latestTimestamp: number;
+    protocolVersion: string;
+    round: bigint;
```

Comment:
I wonder if we should make the `uint64` and `integer` API types map to `number | bigint`? They should both get encoded the same way by the SDK, so I think this change would make the types more flexible to "input" types like this that users create.

Reply:
Yes, I think that makes sense.